src/nvim/tui/input.c

```diff
@@ -48,6 +48,21 @@ void tinput_init(TermInput *input, Loop *loop)
 
   int curflags = termkey_get_canonflags(input->tk);
   termkey_set_canonflags(input->tk, curflags | TERMKEY_CANON_DELBS);
+#ifdef WIN32
+  if (!os_isatty(0)) {
+      const HANDLE conin_handle = CreateFile("CONIN$",
+                                             GENERIC_READ | GENERIC_WRITE,
+                                             FILE_SHARE_READ | FILE_SHARE_WRITE,
+                                             (LPSECURITY_ATTRIBUTES)NULL,
+                                             OPEN_EXISTING, 0, (HANDLE)NULL);
```

Comment:
is `FILE_SHARE_WRITE` needed? We do not specify it in https://github.com/neovim/neovim/blob/16b1e8f9c070ad853c6c63b43591e297bf512662/src/nvim/main.c#L1136